File: TransCoder_tokenized_test_set_functions/python/CHECK_IF_STACK_ELEMENTS_ARE_PAIRWISE_CONSECUTIVE.py
def pairWiseConsecutive ( s ) :
    aux = [ ]
    while ( len ( s ) != 0 ) :
        aux . append ( s [ - 1 ] )
        s . pop ( )
    result = True
    while ( len ( aux ) > 1 ) :
        x = aux [ - 1 ]
        aux . pop ( )
        y = aux [ - 1 ]
        aux . pop ( )
        if ( abs ( x - y ) != 1 ) :
            result = False
        s . append ( x )
        s . append ( y )
    if ( len ( aux ) == 1 ) :
        s . append ( aux [ - 1 ] )
    return result

File: TransCoder_tokenized_test_set_functions/java/CHECK_IF_STACK_ELEMENTS_ARE_PAIRWISE_CONSECUTIVE.java
static boolean pairWiseConsecutive ( Stack < Integer > s ) {
  Stack < Integer > aux = new Stack < Integer > ( );
  while ( ! s . isEmpty ( ) ) {
    aux . push ( s . peek ( ) );
    s . pop ( );
  }
  boolean result = true;
  while ( aux . size ( ) > 1 ) {
    int x = aux . peek ( );
    aux . pop ( );
    int y = aux . peek ( );
    aux . pop ( );
    if ( Math . abs ( x - y ) != 1 ) result = false;
    s . push ( x );
    s . push ( y );
  }
  if ( aux . size ( ) == 1 ) s . push ( aux . peek ( ) );
  return result;
}